pkg/controller/instance/instance_controller.go

```diff
@@ -182,9 +168,11 @@ func (r *ReconcileInstance) Reconcile(request reconcile.Request) (reconcile.Resu
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+
 		if harbor.Status.SpecHash != specHash {
 			harbor.Status.Phase.Name = registriesv1alpha1.InstanceStatusPhaseInstalling
 			harbor.Status.SpecHash = specHash
+			helper.PushFinalizer(harbor, FinalizerName)
```

Comment:
Adding a finalizer when an harbor instance object is in ready state ensures the helm deletion of successfully installed objects, but what about failed cases? For example the installation via helm is triggered but for arbitrary circumstances failed. You may wan't to trigger a finalizer there too, when deleting the instance object, to clean up the helm installation.
As I understand it, it would leave the helm "mess" untouched, because no finalizer has been added there. Might be wrong though. @martin-helmich @elenz97 what you think about that?

Reply:
I'm with you on that. While helm itself has no finalizers, we still need to manipulate helm releases, no matter what their status is. Setting the finalizer at `InstanceStatusPhaseInstalling` (when an instance object has been created) would get us a way more consistent experience using the operator :+1:

Reply:
:+1: on this. IMO it makes sense to set the finalizer as soon as there is something to clean up (that isn't automatically removed by Kubernetes). If the Helm release might need cleaning up when the Phase switches to `Installing`, that's also when the finalizer should be added.

Reply:
ok, the finalizers are now added just before the installation/creation steps